contracts/root/RootChain.sol

```diff
@@ -161,19 +165,42 @@ contract RootChain is Ownable, WithdrawManager {
     stakeManager = StakeManager(_stakeManager);
   }
 
-  function submitHeaderBlock(bytes32 root, uint256 end, bytes sigs) public {
-    uint256 start = currentChildBlock();
+  function getSha256(bytes input) public view returns (bytes20) {
+    bytes32 hash = sha256(input);
+    return bytes20(hash);
+  }
+
+  function submitHeaderBlock(bytes vote, bytes sigs, bytes extradata) public {
+
+    RLP.RLPItem[] memory dataList = vote.toRLPItem().toList();
+    require(keccak256(dataList[0].toData()) == chain, "Chain ID not same");
+    require(keccak256(dataList[1].toData()) == keccak256(roundType), "Round Type Not Same ");
+    require(keccak256(dataList[5].toData()) == keccak256(voteType), "Vote Type Not Same");
+
+    // validate extra data using getSha256(extradata) 
+    require(keccak256(dataList[6].toData()) == keccak256(getSha256(extradata)));
+    // check proposer 
+    address proposer = dataList[5].toAddress();
+    require(msg.sender == proposer);
+
+    // extract end and assign to current child 
+    RLP.RLPItem[] memory txDataList = extradata.toRLPItem().toList()[0].toList();
+    uint256 start = currentChildBlock(); 
+    uint256 end = txDataList[2].toUint();
```

Comment:
i think we can remove end and root variables and address them directly using txDataList ? right ?

Reply:
Trye.